core/service.go

```diff
@@ -123,9 +122,9 @@ func NewService(cfg *Config) (*Service, error) {
 		// create a new BABE session
 		bs, err := babe.NewSession(bsConfig)
 		if err != nil {
-			log.Error("[core] could not start babe session", "error", err)
 			srv.isBabeAuthority = false
-			return nil, nil
+			log.Error("[core] could not start babe session", "error", err)
+			return srv, nil
```

Comment:
return nil err after babe.NewSession fails, why is that ?

Reply:
if we return an error here, then it gets propogated up to configcmd.go `makeNode` which will cause the node to shutdown. it's not necessarily a fatal error to not be able to start BABE, since the other parts of the node can still run. 

it depends on what we want the behaviour of the node to look like, if we are trying to run an authority node and we can't start BABE, should the node exit completely or should it just log an error?

Reply:
logging the error and continuing to start the node seems ok for now

Reply:
cool, I'll leave it as is then, if that's good with everyone!

Reply:
well, I would vote to then check if babe was requested, if so, return the err.

Reply:
I've updated it